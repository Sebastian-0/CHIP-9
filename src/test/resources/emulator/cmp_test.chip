# Test simple xors
ldi A, 5
ldi B, 5
ldi C, 5
ldi D, 5
ldi E, 5
ldi H, 5
ldi L, 5
ldi (HL), 5

# All are equal
clrflag
cmp A
sout FLAGS
clrflag
cmp B
sout FLAGS
clrflag
cmp C
sout FLAGS
clrflag
cmp D
sout FLAGS
clrflag
cmp E
sout FLAGS
clrflag
cmp H
sout FLAGS
clrflag
cmp L
sout FLAGS
clrflag
cmp (HL)
sout FLAGS

# All are smaller
ldi A, 6

clrflag
cmp A
sout FLAGS
clrflag
cmp B
sout FLAGS
clrflag
cmp C
sout FLAGS
clrflag
cmp D
sout FLAGS
clrflag
cmp E
sout FLAGS
clrflag
cmp H
sout FLAGS
clrflag
cmp L
sout FLAGS
clrflag
cmp (HL)
sout FLAGS

# All are larger
ldi A, 4

clrflag
cmp A
sout FLAGS
clrflag
cmp B
sout FLAGS
clrflag
cmp C
sout FLAGS
clrflag
cmp D
sout FLAGS
clrflag
cmp E
sout FLAGS
clrflag
cmp H
sout FLAGS
clrflag
cmp L
sout FLAGS
clrflag
cmp (HL)
sout FLAGS

hcf