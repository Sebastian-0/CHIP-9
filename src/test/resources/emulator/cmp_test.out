160
160
160
160
160
160
160
160

160
80
80
80
80
80
80
80

160
32
32
32
32
32
32
32